testsuites/basic/timesync.py

```diff
@@ -0,0 +1,69 @@
+import re
+
+from assertpy import assert_that
+
+from lisa import Environment, Node, TestCaseMetadata, TestSuite, TestSuiteMetadata
+from lisa.operating_system import Windows
+from lisa.testsuite import simple_requirement
+from lisa.tools import Cat, Dmesg
+from lisa.util import get_matched_str
+
+
+@TestSuiteMetadata(
+    area="time",
+    category="functional",
+    description="""
+    This test suite is related with time sync.
+    """,
+    tags=["timesync"],
+)
+class TimeSync(TestSuite):
+    ptp_registered_dmesg_pattern = re.compile(r"PTP clock support registered", re.M)
+    hyperv_ptp_udev_rule_pattern = re.compile(r"ptp_hyperv", re.M)
+    chrony_path = ["/etc/chrony.conf", "/etc/chrony/chrony.conf"]
+
+    @TestCaseMetadata(
+        description="""
+        https://docs.microsoft.com/en-us/azure/virtual-machines/linux/time-sync#check-for-ptp-clock-source # noqa: E501
+        This test is to check -
+            1. PTP time source is available on Azure guests (newer versions of Linux).
+            2. PTP device name is hyperv.
+            3. When AN enabled, muiltple PTP devices will be available,
+             the names of ptp are changeable, create the symlink /dev/ptp_hyperv to whichever
+             /dev/ptp entry corresponds to the Azure host.
+            4. Chrony should be configured to use this symlink instead of
+             /dev/ptp0 or /dev/ptp1.
+        """,
+        priority=2,
+        requirement=simple_requirement(unsupported_os=[Windows]),
+    )
+    def timesync_validate_ptp(self, environment: Environment, node: Node) -> None:
+        # 1. PTP time source is available on Azure guests (newer versions of Linux).
+        dmesg = node.tools[Dmesg]
+        result = get_matched_str(
```

Comment:
Because the matched ones are string, not regexp, so try built-in methods of assertpy.